runtime/src/accounts_db.rs

```diff
@@ -2886,7 +3321,8 @@ impl AccountsDB {
     }
 
     pub fn add_root(&self, slot: Slot) {
-        self.accounts_index.add_root(slot)
+        self.accounts_index.add_root(slot);
+        self.accounts_cache.add_root(slot);
```

Comment:
Is there any difference between the accounts_cache and accounts_index root set?

Reply:
@sakridge, the roots added should be the same via `add_root`, the difference is that they're cleared differently, i.e. the AccountsCache one is used to track unflushed roots, and flushed here: https://github.com/solana-labs/solana/blob/e38d01632d3c837ffe8cf9963591526dca816b84/runtime/src/accounts_db.rs#L2363-L2373.

The alternative is to track the last `max_flushed_index`, then lock/iterate through the existing `accounts_index` roots set and clone out the ones > `max_flushed_index`, which wouldn't save significant memory, so it seemed easier to avoid the data dependency and just track a separate unflushed work queue.